manager/integration/tests/test_csi.py

```diff
@@ -536,3 +536,119 @@ def test_csi_expansion_with_replica_failure(client, core_api, storage_class, pvc
     write_pod_volume_data(core_api, pod_name, test_data)
     resp = read_volume_data(core_api, pod_name)
     assert resp == test_data
+
+@pytest.mark.skip(reason="TODO") # NOQA
+def test_csi_volumesnapshot_basic():  # NOQA
+    """
+    Test creation / restoration / deletion of a backup via the csi snapshotter
+
+    Context:
+
+    We want to allow the user to programmatically create/restore/delete longhorn backups
+    [csi snapshot mechanism](https://kubernetes.io/docs/concepts/storage/volume-snapshots/)
+
+    Setup:
+
+    1. Make sure your cluster contains the below crds
+    https://github.com/kubernetes-csi/external-snapshotter/tree/master/client/config/crd
+    2. Make sure your cluster contains the snapshot controller
+    https://github.com/kubernetes-csi/external-snapshotter/tree/master/deploy/kubernetes/snapshot-controller
+
+    Steps:
+
+    def csi_volumesnapshot_creation_test(snapshotClass=longhorn|custom):
+    1. create volume(1)
+    2. write data to volume(1)
+    3. Create a kubernetes `VolumeSnapshot` object with a test uuid
+    4. The `VolumeSnapshot.uuid` will be used to identify a **longhorn snapshot**
```

Comment:
Unclear what should the test do at this step.

Reply:
Thanks the test is supposed to create a VolumeSnapshot object with a test choosen uuid so it can do the lookup easily afterwards. I will reword the phrasing.